### PR TITLE
Add two more validators, providing four nodes in a devnet.

### PR DESCRIPTION
**Pull request by vitropy**
Seems healthy? I'm not sure how to assign a validator to a signing group but I don't think we did anything special to make that happen with a two-node setup or if that's required with this one. The logs do look healthy, though. Examine the logs with a command such as this:

```sh
cd $ENTROPY_WORKSPACE_DIR/sdk
docker compose --file dev/docker-scripts/four-nodes.yaml logs --follow
```

Use this file in CI or in your workflow with a command like:

```sh
cd $ENTROPY_WORKSPACE_DIR/sdk
./dev/bin/spin-up.sh four-nodes
```

### CODE REVIEW

**Review by rh0delta — DISMISSED** [review]

**Comment by mixmix** [comment]
@vitropy are you leading this PR?
Whoever leads it will like need to collaborate with you/ core/ js?

Checklist before this is ready:
- [ ] tests passing in CI
- [ ] review

**Comment by mixmix** [comment]
@vitropy you opened this PR... but also Frankie committed to it.

What I was pointing to with the "leading this PR" request is that from prior experience on other teams, if it's unclear who is doing a task / is responsible for getting it done, then it is more inclined to slip through the cracks and get forgotten.

The culture I've found useful in the past is to make it clear who the leader / main backstop person is. We can make up whatever protocols we want of course!

What maybe I could have done is asked instead:
- what is the state of this PR? (in-progress / old and should be closed / on hold / other)
- if it's still in progress, who is progressing it?
- is there any support needed here?

**Comment by JesseAbram** [comment]
Created a chain type to support the docker file, I guess next steps here are to check (test this docker file)

File: dev/docker-scripts/four-nodes.yaml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+  alice-chain-node:
+    image: entropyxyz/entropy:${ENTROPY_CORE_VERSION:-latest}
+    ports:
+      - "127.0.0.1:9944:9944/tcp"
+    volumes:
+      - ${PWD}/dev:/srv/entropy/dev
+    command:
+      - "--chain"
+      - "devnet-local"
```
Comment (JesseAbram) [inline dev/docker-scripts/four-nodes.yaml:39]: all of these chain arguments should now be ```testnet-local-large```

File: dev/docker-scripts/four-nodes.yaml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  # "Bob's chain node."
+  bob-chain-node:
+    image: entropyxyz/entropy:${ENTROPY_CORE_VERSION:-latest}
+    ports:
+      - "127.0.0.1:9945:9944/tcp"
+    volumes:
+      - ${PWD}/dev:/srv/entropy/dev
+    command:
+      - "--chain"
+      - "devnet-local"
```
Comment (JesseAbram) [inline dev/docker-scripts/four-nodes.yaml:73]: testnet-local-large

File: dev/docker-scripts/four-nodes.yaml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+  dave-chain-node:
+    image: entropyxyz/entropy:${ENTROPY_CORE_VERSION:-latest}
+    ports:
+      - "127.0.0.1:9946:9944/tcp"
+    volumes:
+      - ${PWD}/dev:/srv/entropy/dev
+    command:
+      - "--chain"
+      - "devnet-local"
```
Comment (JesseAbram) [inline dev/docker-scripts/four-nodes.yaml:107]: testnet-local-large

File: dev/docker-scripts/four-nodes.yaml [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+  eve-chain-node:
+    image: entropyxyz/entropy:${ENTROPY_CORE_VERSION:-latest}
+    ports:
+      - "127.0.0.1:9947:9944/tcp"
+    volumes:
+      - ${PWD}/dev:/srv/entropy/dev
+    command:
+      - "--chain"
+      - "devnet-local"
```
Comment (JesseAbram) [inline dev/docker-scripts/four-nodes.yaml:141]: testnet-local-large

File: dev/docker-scripts/four-nodes.yaml (hunk 1 above)
Comment (mixmix) [reply dev/docker-scripts/four-nodes.yaml:39]:
```suggestion
      - "testnet-local-large"
```

File: dev/docker-scripts/four-nodes.yaml (hunk 2 above)
Comment (mixmix) [reply dev/docker-scripts/four-nodes.yaml:73]:
```suggestion
      - "testnet-local-large"
```

File: dev/docker-scripts/four-nodes.yaml (hunk 3 above)
Comment (mixmix) [reply dev/docker-scripts/four-nodes.yaml:107]:
```suggestion
      - "testnet-local-large"
```

File: dev/docker-scripts/four-nodes.yaml (hunk 4 above)
Comment (mixmix) [reply dev/docker-scripts/four-nodes.yaml:141]:
```suggestion
      - "testnet-local-large"
```

**Comment by mixmix** [comment]
> Like most things, this is a collaborative effort.

Yeah I know, just trying to figure out what our patterns of collaboration look like. I've been caught out before assuming "sensible process". e.g. Assignee field is a tool, AND not everyone uses it the same.

I think what you've proposed is good and clear.

I've added a little help by
- resolving a merge conflict (updated this branch from `main`)
- adding "suggestions" to make it easier to do those changes Jesse recommended

_I'll check in each week to see if there's anything that I can help with - you're assigned here but this is our lounge and want it to go well for you :heart:_

**Comment by mixmix** [comment]
Changes in core, need to work on getting the RC of core 0.2.0 tested, deployed and a new release of SDK  which uses that

image

**Comment by mixmix** [comment]
@frankiebee can you process this <3

**Comment by frankiebee** [comment]
Closing this due to inactivity.

File: dev/docker-scripts/four-nodes.yaml [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+  alice-chain-node:
+    image: entropyxyz/entropy:${ENTROPY_CORE_VERSION:-latest}
+    ports:
+      - "127.0.0.1:9944:9944/tcp"
+    volumes:
+      - ${PWD}/dev:/srv/entropy/dev
+    command:
+      - "--chain"
+      - "devnet-local-four-nodes"
```
Comment (JesseAbram) [inline dev/docker-scripts/four-nodes.yaml:39]: all these need to change to ```devnet-local```

Comment (JesseAbram) [reply dev/docker-scripts/four-nodes.yaml:39]: https://github.com/entropyxyz/entropy-core/blob/master/node/cli/src/command.rs#L80

**Pull request closed without merging**